src/Telegram.Bot/Types/Color.cs

```diff
@@ -0,0 +1,179 @@
+    /// <param name="blue">Blue component</param>
+    public Color(int red, int green, int blue)
+    {
+        CheckByte(red);
+        CheckByte(green);
+        CheckByte(blue);
+        (Red, Green, Blue) = (red, green, blue);
+    }
+
+    /// <summary>
+    /// Instantiate a new color value
+    /// </summary>
+    /// <param name="red">Red component</param>
+    /// <param name="green">Green component</param>
+    /// <param name="blue">Blue component</param>
+    public Color(uint red, uint green, uint blue)
+    {
+        CheckByte(red);
+        CheckByte(green);
+        CheckByte(blue);
+        (Red, Green, Blue) = ((int)red, (int)green, (int)blue);
+    }
+
+    /// <summary>
+    /// Instantiate a new color value
+    /// </summary>
+    /// <param name="color">Numeric value of color in RGB space</param>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public Color(int color)
+    {
+        if (color is > MaxRgbValue or < 0)
+        {
+            throw new ArgumentOutOfRangeException(nameof(color), color, "Color is out of range");
+        }
+
+        Red = color >> RedShift;
+        Green = color >> GreenShift;
+        Blue = color >> BlueShift;
+    }
+
+    /// <summary>
+    /// Instantiate a new color value
+    /// </summary>
+    /// <param name="color">Numeric value of color in RGB space</param>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public Color(uint color)
+    {
+        if (color is > MaxRgbValue or < 0)
+        {
+            throw new ArgumentOutOfRangeException(nameof(color), color, "Color is out of range");
+        }
+
+        Red = unchecked((int)color >> RedShift);
+        Green = unchecked((int)color >> GreenShift);
+        Blue = unchecked((int)color >> BlueShift);
+    }
+
+    public override string ToString() => $"#{ToInt():X24}";
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns>Numeric representation of current color</returns>
+    public int ToInt()
+    {
+#if NETCOREAPP3_1_OR_GREATER
+        ReadOnlySpan<byte> components = stackalloc [] { (byte)Red,(byte)Green, (byte)Blue };
+        return BitConverter.ToInt32(components);
+#else
+        byte[] components = { (byte)Red,(byte)Green, (byte)Blue };
+        return BitConverter.ToInt32(components, 0);
+#endif
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns></returns>
+    public byte[] ToBytes() => BitConverter.GetBytes(ToInt());
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="color"></param>
+    /// <returns></returns>
+    public static explicit operator int(Color color) => color.ToInt();
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="color"></param>
+    /// <returns></returns>
+    public static explicit operator byte[](Color color) => color.ToBytes();
+
+    /// <summary>
+    /// Blue color
+    /// </summary>
+    public static Color BlueColor = new(0x6FB9F0);
```

Comment:
maybe **Blue** is enough? Color.BlaBlaColor IMO is a bit repetitive.

Reply:
I thought so too, but it clashes with instance member Blue :(

Reply:
Another solution is to instead add some suffix to the components, e.g. `Color {RedComponent, GreenComponent, BlueComponent }`.